### PR TITLE
Bug 1881027 - Cluster installation fails at with error : the containe…

### PR DESCRIPTION
**Pull request by eranco74**
…r name \"assisted-installer\" is already in use.

Failure is caused by executing multiple install commands on the host.
Since only one container with the name "assisted-installer" can be executed just one install command is actually running and the rest return with error setting the host installation to failure
The trigger to this behavior is the time it take to pull the installer image from the downstream registry.
This patch mitigate the error by not moving the host to error in case the install command fail with that specific error code and stderr output (the container name \"assisted-installer\" is already in use) .

### CODE REVIEW

**Comment by eranco74** [comment]
Adding tests

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1521,7 +1522,7 @@ func (b *bareMetalInventory) PostStepReply(ctx context.Context, params installer
 	if params.Reply.ExitCode != 0 {
 		err = fmt.Errorf(msg)
 		log.WithError(err).Errorf("Exit code is <%d> ", params.Reply.ExitCode)
-		handlingError := handleReplyError(params, b, ctx, &host)
+		handlingError := b.handleReplyError(params, ctx, &host)
```
Comment (filanov) [inline internal/bminventory/inventory.go:1525]: thank you

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -1551,9 +1552,14 @@ func (b *bareMetalInventory) PostStepReply(ctx context.Context, params installer
 	return installer.NewPostStepReplyNoContent()
 }
 
-func handleReplyError(params installer.PostStepReplyParams, b *bareMetalInventory, ctx context.Context, h *models.Host) error {
+func (b *bareMetalInventory) handleReplyError(params installer.PostStepReplyParams, ctx context.Context, h *models.Host) error {
 
 	if params.Reply.StepType == models.StepTypeInstall {
+		// Handle case of installation failure due to an already running assisted-installer.
+		if params.Reply.ExitCode == 137 && strings.Contains(params.Reply.Error, "the container name \"assisted-installer\" is already in use"){
+			b.log.Infof("Install command failed due to an already running installation: %s", params.Reply.Error)
```
Comment (filanov) [inline internal/bminventory/inventory.go:1560]: Suggesting to make it a warning.

**Comment by eranco74** [comment]
See https://github.com/openshift/assisted-service/pull/407

**Pull request closed without merging**